### PR TITLE
UCT/MM: enhance the error message from shm_open/open.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by alinask** [comment]
related to https://github.com/openucx/ucx/issues/856

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2257/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1582/ for details (Mellanox internal link).

File: src/uct/sm/mm/mm_posix.c [hunk 1]
```diff
@@ -429,8 +429,14 @@ static ucs_status_t uct_posix_attach(uct_mm_id_t mmid, size_t length,
     }
 
     if (shm_fd == -1) {
-        ucs_error("Error returned from shm_open/open in attach. %m. File name is: %s",
-                  file_name);
+        if ((mmid & UCT_MM_POSIX_PROC_LINK) ||
+            (!(mmid & UCT_MM_POSIX_PROC_LINK) && !(mmid & UCT_MM_POSIX_SHM_OPEN))) {
```
Comment (yosefe) [inline src/uct/sm/mm/mm_posix.c]: redundant condition - could re write it as:
```
 ucs_error("Error returned from open in attach. %m. File name is: %s%s",
           (mmid & UCT_MM_POSIX_PROC_LINK) ? "" :
           (mmid & UCT_MM_POSIX_SHM_OPEN) ? "/dev/shm" :
           "",
           file_name)
```

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2264/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1592/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1594/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2267/ for details.

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2268/ for details.

**Comment by yosefe** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1600/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2273/ for details.

**Comment by yosefe** [comment]
failure is unrelated

**Pull request merged**